docs/en/engines/table-engines/special/aggregatingmemory.md

```diff
@@ -0,0 +1,74 @@
+---
+toc_priority: 45
+toc_title: AggregatingMemory
+---
+
+# AggregatingMemory {#aggregatingmemory}
+
+The `AggregatingMemory` is a special engine, which allows incrementally aggregate data, keeping it all in the memory. When writing to this table, incoming data is aggregated as specified in the AS `SELECT` query, and internal memory state is updated. Every `INSERT` to this table is processed incrementally, and updates internal state without iterating over previously aggregated data. `SELECT` from the table returns current aggregation state, and can be called any amount of times.
+
+When restarting a server, data disappears from the table and the table becomes empty. 
+
+You can use `AggregatingMemory` tables for incremental data aggregation, including for aggregated materialized views.
+
+## Creating a Table {#creating-a-table}
+
+``` sql
+CREATE TABLE [IF NOT EXISTS] [db.]table_name
+ENGINE = AggregatingMemory()
+AS SELECT
```

Comment:
This work looks awesome. May i ask if we are going to support nested subquery in `AS SELECT` like 

```
CREATE TABLE [db.]table_name
ENGINE = AggregatingMemory()
AS SELECT count(*) FROM (
        SELECT avg(column) AS a FROM src_table
    )   WHERE a > 123
```

Reply:
Looks like it does not work for subqueries so far.